subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/caching/ScriptBuildCacheKeyTest.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.caching
+
+import org.gradle.kotlin.dsl.cache.ScriptBuildCacheKey
+
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+
+import org.junit.Test
+
+
+class ScriptBuildCacheKeyTest {
+
+    @Test
+    fun `it removes invalid characters from the cache key prefix`() {
+
+        assertThat(
+            buildCacheKeyFor("gradle-kotlin-dsl/dx7phw87epyvb7zje5dti3tvl"),
```

Comment:
So I think I understand why the build cache keys look like this: You are using `CacheKeyBuilder.build` which returns the prefix + compactString for the hash: https://github.com/gradle/gradle/blob/500c8083512a9011c420a612eede6cbe29c43542/subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultCacheKeyBuilder.java#L58

Wouldn't it be better to expose the combined hash from there directly, instead of doing the de-tour via the `String`?

Could you tell me what things go into the `CacheKeySpec` for the build script or where to find that information?

Reply:
@wolfs Yes, the build cache keys look like that because of `CacheKeyBuilder.build` and because @eskatos and I thought it would be good to keep the prefix for the informational value and key space partitioning it brings.

> Wouldn't it be better to expose the combined hash from there directly, instead of doing the de-tour via the String?

Possibly.

The script cache key is computed as:

https://github.com/gradle/kotlin-dsl/blob/30f8adc96bb2fe6fc397d922d677c948b8486ceb/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinScriptEvaluator.kt#L228

Plus the generated accessors classpath if any.

Reply:
@wolfs we decided to discard the prefix for the remote build cache key

Reply:
Good. We are still using a compacted md5 hash string, so the keys used for kotlin caching will be quite different than the rest. I think that should be fixed for 5.0. WDYT?